pulser/sequence.py

```diff
@@ -133,39 +140,46 @@ def __init__(self, register, device):
         # Checks if register is compatible with the device
         device.validate_register(register)
 
-        self._register = register
-        self._device = device
-        self._in_xy = False
+        self._register: Register = register
+        self._device: Device = device
+        self._in_xy: bool = False
         self._calls = [_Call("__init__", (register, device), {})]
-        self._channels = {}
-        self._schedule = {}
-        self._phase_ref = {}  # The phase reference of each channel
+        self._channels: Dict[str, Channel] = {}
+        self._schedule: Dict[str, List[_TimeSlot]] = {}
+        # The phase reference of each channel
+        self._phase_ref: Dict[str, Dict[str, _PhaseTracker]] = {}
```

Comment:
On second thought, it would be best to use `TypeVar` to define a type for qubit ID's and use it throughout

Reply:
> There might be a minor issue here with the type of the qubits, as it doesn't strictly have to be a `str`, they can be `int` or `float` too (as long as they serve as keys for `dict`).

I think `float` should not be allowed as keys of `dict` to identify qubits. It does not make much sense. And it seems to result in a lot of troubles.

Reply:
I agree, I just mentioned it because it can indeed be used as a dictionary key, but I'm in favour of not allowing it explicitly, just `int` and `str`. Sorry for the mix-up